components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/util/APIFileUtils.java

```diff
@@ -332,19 +331,14 @@ public static void exportThumbnailToFileSystem(InputStream thumbnailInputStream,
      * @throws APIMgtDAOException if an error occurs while writing to the file
      */
     public static void writeStreamToFile(String path, InputStream inputStream) throws APIMgtDAOException {
-        FileOutputStream outputStream = null;
-        try {
-            outputStream = new FileOutputStream(path);
+        try (FileOutputStream outputStream = new FileOutputStream(path)) {
             IOUtils.copy(inputStream, outputStream);
-        } catch (FileNotFoundException e) {
-            throw new APIMgtDAOException(e);
-
         } catch (IOException e) {
-            throw new APIMgtDAOException("Unable to write to file at path: " + path, e);
-
+            String errorMsg = "Unable to write to file at path: " + path;
+            log.error(errorMsg, e);
+            throw new APIMgtDAOException(errorMsg, e);
         } finally {
             IOUtils.closeQuietly(inputStream);
```

Comment:
InputStream is passed as a parameter from REST API at [1]. We have to close it manually after writing the file. Therefore we can't use try with resources here.

[1] https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java#L1018